lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -35,6 +40,47 @@ def __init__(self, apis, concrete_artifacts_manager):
         self._apis = apis
         self._concrete_art_mgr = concrete_artifacts_manager
 
+    def _get_collection_versions(self, requirement):
+        # type: (Requirement, Iterator[GalaxyAPI]) -> Iterator[Tuple[GalaxyAPI, str]]
```

Comment:
It seems like this change (and maybe some of the previous ones) break the typing check (which is still only present in an unmerged PR atm). I'll try to look into this more closely.

Reply:
:eyes:

Reply:
Oh, I see. It describes two args while accepting only one. It doesn't accept an iterator returning galaxy apis.

Reply:
Opened a PR if you want to review #76296

Reply:
Oh, I was working on my own patch today.

Reply:
Ah, okay. I took the comment as a cue that you wanted me to fix it.